extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -115,14 +115,17 @@ async function databaseArchiveFetcher(
 }
 
 async function getStorageFolder(storagePath: string, urlStr: string) {
+  // we need some approximation for a folder name
```

Comment:
Not clear to me what's being approximated?

Reply:
I'll make that clearer. The max file name length is 260 on windows and 255 on macs. But the zip files tend to have very long names. Also, we add a few extra chars at the end in case there are duplicates.

I'm partially thinking that we should just avoid this and use a random name.

Reply:
Wait...we can't just use a random name since we also use the folder name as the database name when we import the database.